task8/src/java/AdCollection.java

```diff
@@ -0,0 +1,171 @@
+package com;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.List;
+
+public class AdCollection {
+    private List<Ad> adList;
+
+    public AdCollection() {
+        this.adList = new ArrayList<Ad>();
+    }
+
+    public AdCollection(List<Ad> adList) {
+        this.adList = adList;
+    }
+
+    //getAds(1,4,"vendor:Adobe");
+    public List<Ad> getAds(int skip, int top, String filter) {
+
+        List<Ad> resultAds;
+
+        if (filter != null) {
+            String[] filterOptions = filter.split(":");
```

Comment:
In production, you would need a check that there is something actually present after `:` in the string